frontend/src/pages/ProductDetail/ProductDetail.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { AiFillStar } from 'react-icons/ai';
+import classNames from 'classnames';
+import renderProductPrice from '@assets/helper/products';
+import { addToCart } from '@components/common/Header/Cart/cartSlice';
+import useProductDetail from '@hooks/useProductDetail';
+import { DEFAULT_PRODUCT } from '@pages/Home/ProductList/Product/Product';
+import Selection from './Selection/Selection';
+import './ProductDetail.scss';
+
+const ProductDetail = () => {
+  const dispatch = useDispatch();
+  const productName = useParams().name || '';
+  const { product } = useProductDetail(productName);
```

Comment:
this should be `const { product = DEFAULT_PRODUCT } = useProductDetail(productName);`, and you can remove `||` in line `17`